src/EnergyPlus/Plant/DataPlant.hh

```diff
@@ -92,126 +89,217 @@ namespace DataPlant {
+        "Pump:ConstantSpeed",
+        "Pump:VariableSpeed:Condensate",
+        "HeaderedPumps:VariableSpeed",
+        "HeaderedPumps:ConstantSpeed",
+        "WaterUse:Connections",
+        "Coil:Cooling:Water",
+        "Coil:Cooling:Water:DetailedGeometry",
+        "Coil:Heating:Water",
+        "Coil:Heating:Steam",
+        "Solarcollector:Flatplate:Water",
+        "LoadProfile:Plant",
+        "GroundHeatExchanger:System",
+        "GroundHeatExchanger:Surface",
+        "GroundHeatExchanger:Pond",
+        "Generator:Microturbine",
+        "Generator:InternalCombustionEngine",
+        "Generator:CombustionTurbine",
+        "Generator:Microchp",
+        "Generator:Fuelcell:StackCooler",
+        "FluidCooler:SingleSpeed",
+        "FluidCooler:TwoSpeed",
+        "EvaporativeFluidCooler:SingleSpeed",
+        "EvaporativeFluidCooler:TwoSpeed",
+        "ThermalStorage:ChilledWater:Mixed",
+        "ThermalStorage:ChilledWater:Stratified",
+        "SolarCollector:FlatPlate:PhotovoltaicThermal",
+        "ZoneHVAC:Baseboard:Convective:Water",
+        "ZoneHVAC:Baseboard:RadiantConvective:Steam",
+        "ZoneHVAC:Baseboard:RadiantConvective:Water",
+        "ZoneHVAC:LowTemperatureRadiant:VariableFlow",
+        "ZoneHVAC:LowTemperatureRadiant:ConstantFlow",
+        "AirTerminal:SingleDuct:ConstantVolume:CooledBeam",
+        "Coil:Heating:WaterToAirHeatPump:EquationFit",
+        "Coil:Cooling:WaterToAirHeatPump:EquationFit",
+        "Coil:Heating:WaterToAirHeatPump:ParameterEstimation",
+        "Coil:Cooling:WaterToAirHeatPump:ParameterEstimation",
+        "Refrigeration:Condenser:WaterCooled",
+        "Refrigeration:CompressorRack",
+        "AirLoopHVAC:UnitaryHeatPump:AirToAir:MultiSpeed",
+        "ChillerHeater:Absorption:DoubleEffect",
+        "PipingSystem:Underground:PipeCircuit",
+        "SolarCollector:IntegralCollectorStorage",
+        "Coil:Heating:WaterToAirHeatPump:VariableSpeedEquationFit",
+        "Coil:Cooling:WaterToAirHeatPump:VariableSpeedEquationFit",
+        "PlantComponent:UserDefined",
+        "Coil:UserDefined",
+        "ZoneHVAC:ForcedAir:UserDefined",
+        "AirTerminal:SingleDuct:UserDefined",
+        "AirConditioner:VariableRefrigerantFlow",
+        "GroundHeatExchanger:HorizontalTrench",
+        "HeatExchanger:FluidToFluid",
+        "PlantComponent:TemperatureSource",
+        "CentralHeatPumpSystem",
+        "AirLoopHVAC:UnitarySystem",
+        "Coil:Cooling:DX:SingleSpeed:ThermalStorage",
+        "CoolingTower:VariableSpeed:Merkel",
+        "SwimmingPool:Indoor",
+        "GroundHeatExchanger:Slinky",
+        "WaterHeater:HeatPump:WrappedCondenser",
+        "AirTerminal:SingleDuct:ConstantVolume:FourPipeBeam",
+        "ZoneHVAC:CoolingPanel:RadiantConvective:Water",
+        "HeatPump:PlantLoop:EIR:Cooling",
+        "HeatPump:PlantLoop:EIR:Heating"};
+
+    constexpr std::array<std::string_view, static_cast<size_t>(PlantEquipmentType::Num)> PlantEquipTypeNames{
```

Comment:
Similarly `constexpr std::array<std::string_view> PlantEquipTypeNames` to replace the appropriate strings.